catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/WorkspaceServiceImpl.java

```diff
@@ -28,60 +27,80 @@
 import java.io.Serializable;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
+import javax.annotation.concurrent.ThreadSafe;
 import org.apache.commons.lang3.tuple.ImmutablePair;
 import org.apache.commons.lang3.tuple.Pair;
 import org.codice.ddf.catalog.ui.metacard.workspace.QueryMetacardImpl;
 import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceMetacardImpl;
 import org.codice.ddf.catalog.ui.metacard.workspace.transformer.impl.WorkspaceTransformerImpl;
-import org.codice.ddf.catalog.ui.query.monitor.api.FilterService;
 import org.codice.ddf.catalog.ui.query.monitor.api.SecurityService;
+import org.codice.ddf.catalog.ui.query.monitor.api.SubscriptionsPersistentStore;
+import org.codice.ddf.catalog.ui.query.monitor.api.WorkspaceQueryBuilder;
 import org.codice.ddf.catalog.ui.query.monitor.api.WorkspaceService;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentStore;
 import org.opengis.filter.Filter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@ThreadSafe
 public class WorkspaceServiceImpl implements WorkspaceService {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(WorkspaceServiceImpl.class);
 
-  private final CatalogFramework catalogFramework;
+  private static final String ID_FIELD = "id_txt";
 
-  private final FilterBuilder filterBuilder;
+  private final CatalogFramework catalogFramework;
 
   private final WorkspaceTransformerImpl workspaceTransformer;
 
-  private final FilterService filterService;
-
   private final SecurityService securityService;
 
+  private final PersistentStore persistentStore;
+
+  private final WorkspaceQueryBuilder workspaceQueryBuilder;
+
+  /** Use an AtomicInteger to make the class thread-safe. */
+  private final AtomicInteger maxSubscriptions = new AtomicInteger(0);
```

Comment:
If we're only doing sets and gets on this AtomicInteger it means we could just replace it with a volatile variable I believe.

Reply:
I check the docs on `volatile`

Reply:
confirmed, and changed to `volatile`